tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -3143,14 +3155,43 @@ Status ReplaceSoftplusTanhAndMulWithMish(
   return OkStatus();
 }
 
+bool RuntimeFusionEnabled(const Cluster& cluster) {
+  static bool is_enabled = [&] {
+#if CUDNN_VERSION >= 8400
+    auto devices = cluster.GetDevices();
+    int num_gpus = 0;
+    int num_ampere = 0;
+    for (const auto& d : devices) {
+      if (d.second.type() == "GPU") {
+        num_gpus++;
+        auto cc_it = d.second.environment().find("architecture");
+        if (cc_it != d.second.environment().end()) {
+          double compute_capability = 0.0;
+          if (absl::SimpleAtod(cc_it->second, &compute_capability) &&
+              compute_capability >= 8.0) {
+            num_ampere++;
+          }
+        }
+      }
+    }
+    return CudnnUseFrontend() && CudnnUseRuntime() && num_gpus > 0 &&
+           num_gpus == num_ampere;
```

Comment:
I think we should print a warning somewhere if someone sets USE_RUNTIME but for some reason the runtime is not enabled?  (Either not Ampere, or whatever else?)